### DISCUSSION TITLE
Changing Docker Port

### DISCUSSION THREAD

**Discussion by gboxxx** [mathesar-foundation/mathesar#2830] [User Q&A]
I'm a little confused on how to change the docker port as this is my first dabble in "profiles." 

I'm not sure what port I should be using to access Mathesar. Everything is up and running, just tried the initial ports (8030) as well as adding custom ports but get nothing. **Any advice would be greatly appreciated!** 

- Docker version 20.10.3, build 55f0773 (on Synology NAS if that matters)
- Docker Compose version v2.1.1

Here's the latest docker-compose I've tried. 

```
version: "3.9"

services:
  db:
    profiles: [ "dev", "prod", "test" ]
    image: postgres:13
    container_name: mathesar_db
    environment:
      - POSTGRES_DB=${POSTGRES_DB-mathesar_django}
      - POSTGRES_USER=${POSTGRES_USER-mathesar}
      - POSTGRES_PASSWORD=${POSTGRES_PASSWORD-mathesar}
    # db container is exposed to the other containers, and the host.
    ports:
      - "5428:5428"
    expose:
      - "5428"
    volumes:
      - /volume1/docker/mathesar/postgres:/var/lib/postgresql/data
    healthcheck:
      test:
        [
          "CMD-SHELL",
          "pg_isready -d $${POSTGRES_DB-mathesar_django} -U $${POSTGRES_USER-mathesar}"
        ]
      interval: 5s
      timeout: 1s
      retries: 30
      start_period: 5s
  base-service:
    profiles: [ "web-server" ]
    container_name: mathesar_service
    image: mathesar/mathesar-prod:latest
    environment:
      - MODE=PRODUCTION
      - DEBUG=${DEBUG-False}
      - DJANGO_ALLOW_ASYNC_UNSAFE=true
      - DJANGO_SETTINGS_MODULE=${DJANGO_SETTINGS_MODULE-config.settings.production}
      - ALLOWED_HOSTS=${ALLOWED_HOSTS-*}
      - SECRET_KEY=2gr6ud88x=(p855_5nbj_+7^bw-iz&n7ldqv%94mjaecl+b9=4
      - DJANGO_DATABASE_KEY=default
      - DJANGO_DATABASE_URL=postgres://mathesar:mathesar@mathesar_db:5432/mathesar_django
      - MATHESAR_DATABASES=(mathesar_tables|postgresql://mathesar:mathesar@mathesar_db:5432/mathesar)
      - HTTP_PORT=8030
      - HTTPS_PORT=8921
      - DJANGO_SUPERUSER_PASSWORD=mysuperpassword
    command: ./run.sh
    volumes:
      - /volume1/docker/mathesar/static:/code/static
      - /volume1/docker/mathesar/media:/code/media
  service:
    profiles: [ "prod" ]
    extends: base-service
    image: mathesar/mathesar-prod:latest
    # Volumes are not shared when extending
    volumes:
      - /volume1/docker/mathesar/prod/static:/code/static
      - /volume1/docker/mathesar/prod/media:/code/media
    depends_on:
      db:
        condition: service_healthy
    healthcheck:
      test: curl -f http://localhost:8030
      interval: 10s
      timeout: 5s
      retries: 30
      start_period: 5s
    # On prod, the HTTP port is exposed to other containers, but not the host.
    expose:
      - "8030"
  caddy-reverse-proxy:
    profiles: [ "prod" ]
    image: mathesar/mathesar-caddy:latest
    environment:
      - DOMAIN_NAME=localhost
    # caddy container is exposed to the other containers, and the host.
    ports:
      - "8038:80"
      - "9281:443"
    volumes:
      - /volume1/docker/mathesar/media:/mathesar/media
      - /volume1/docker/mathesar/static:/mathesar/static
      - /volume1/docker/mathesar/caddy_data:/data
      - /volume1/docker/mathesar/caddy_config:/config
    labels:
      - "com.centurylinklabs.watchtower.enable=true"

volumes:
  postgresql_data:
  media:
  static:
```

**Comment by silentninja** [comment]
```
  service:
    ....
    # On prod, the HTTP port is exposed to other containers, but not the host.
    expose:
      - "8030"
```
The port for `service` isn't configurable as the [gunicorn server](https://github.com/centerofci/mathesar/blob/87d2cd785d4ca9fbbd94badea1098b43f8a21d67/run.sh#L6) is configured to listen on port `8000` only. 

The port is exposed only for internal containers using [`expose`](https://docs.docker.com/compose/compose-file/05-services/#expose) so there shouldn't be any conflicts with external services that want to use port `8000`. Is there a reason why you are using port `8030` instead of `8000` as specified in our [docker-compose](https://github.com/centerofci/mathesar/blob/87d2cd785d4ca9fbbd94badea1098b43f8a21d67/docker-compose.yml#L83)? 


Also, I noticed that you have changed the ports of the `db` incorrectly. The port used by the Postgres server is `5432` and not `5428`.

```
services:
  db:
   ...
    ports:
      - "5428:5428" # It should be 5428:5432, the port number on the right-hand side of `:` is the container port
```
Refer https://docs.docker.com/compose/compose-file/05-services/#ports on how to configure the ports

I would make the following changes to your docker-compose

```diff
services:
  db:
   ....
    ports:
-      - "5428:5428"
+      - "5428:5432"
  ...
  base-service:
  ...
    environment:
    # These environment variables are useless.
-      - HTTP_PORT=8030
-      - HTTPS_PORT=8921
  ...
  service:
   ...
    healthcheck:
-      test: curl -f http://localhost:8030
+      test: curl -f http://localhost:8000
   ...
    expose:
-      - "8030"
+      - "8000"
  caddy-reverse-proxy:
   ...
    ports:
      - "8038:80"
      - "9281:443"
   ...

**Comment by gboxxx** [comment]
I changed 8000 because Portainer is on 8000. I also had a conflict with 5432 but you're right i goofed that one! 

Most of the stuff I self-host I've been able to change ports pretty freely as long as all proper spots are updated. This one kinda confused me a bit. Guess I'll just have to free up 8000?

**Reply by silentninja** [reply]
There is no need to free up port 8000. We are not binding the port to the host network, we only expose it(so that the caddy reverse proxy can forward request to the container) which means the port can be accessed only by specifying the container name. So it won’t clash with the Portainer. You only need to configure the caddy ports which are customisable as they bind to the host network

**Comment by gboxxx** [comment]
Awesome, thanks Mukesh - I'll give that a try and report back!

On Thu, Apr 20, 2023 at 8:41 PM Mukesh ***@***.***> wrote:

> There is no need to free up port 8000. We are not binding the port to the
> host network, we only expose it(so that the reverse proxy can forward
> request to the container) which means the port can be accessed only by
> specifying the container name. So it won’t clash with the Portainer running
> in a different container. . You only need to configure the caddy ports
> which are customisable as they bind to the host network
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/centerofci/mathesar/discussions/2830#discussioncomment-5680524>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AXBYUUIFI5N5TUCY5D4QXHTXCHQUDANCNFSM6AAAAAAXE5HRYQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 

Stay awesome,

[image: Company logo]

Vincent Termini VP, Marketing & Automation
Sacramento, CA
(916) 407-4884 | GrubBoxx.com

**Comment by gboxxx** [comment]
Accessing via Caddy ports doesn't work, I'm guessing because of something with Synology. I think I'll just stick to Baserow for now, but will keep an eye on the project.

**Reply by silentninja** [reply]
Apologies for not being able to get it running. I must admit that our current setup using the install script is very opinionated and there is very little documentation help for running it on various setups but we are working on accommodating running Mathesar on various other setups too #2789  which is due to be released in the next 2 weeks. 

Meanwhile, we can assist you with the installation over a synchronous conversation, if you are interested please use https://mathesar.org/free-install.html. It would greatly help us to understand your setup and help resolve the issue.
